SimMuon/RPCDigitizer/src/RPCSimSetUp.cc

```diff
@@ -166,6 +168,213 @@ void RPCSimSetUp::setRPCSetUp(const std::vector<RPCStripNoises::NoiseItem>& vnoi
     n++;
   }
 }
+*/
+
+// New version that can deal with rolls with an arbitrary number of strips, no hardcoded value 96 here
+void RPCSimSetUp::setRPCSetUp(const std::vector<RPCStripNoises::NoiseItem>& vnoise, const std::vector<RPCClusterSize::ClusterSizeItem>& vClusterSize){
+
+  LogDebug ("rpssimsetup")<<"RPCSimSetUp::setRPCSetUp(vector<NoiseItem>, vector<ClusterSizeItem>)"<<std::endl;
+
+  uint32_t detId, current_detId, this_detId;
+  RPCDetId rpcId, current_rpcId, this_rpcId;
+  const RPCRoll * current_roll,* this_roll;
+  unsigned int current_nStrips;
+
+  LogDebug ("rpssimsetup")<<"RPCSimSetUp::setRPCSetUp :: ClusterSizeItem :: begin"<<std::endl;
+  // ### ClusterSizeItem #######################################################                        
+  std::vector<RPCClusterSize::ClusterSizeItem>::const_iterator itCls;
+  int clsCounter(1);
+  std::vector<double> clsVect;
+  // ### loop for New Format (120 entries)
+  for(itCls = vClusterSize.begin(); itCls != vClusterSize.end(); ++itCls){
+    // LogDebug ("rpssimsetup")<<" Push back clustersize = "<<itCls->clusterSize<<std::endl;
```

Comment:
all LogDebug is not compiled in regular builds. If the commented out debugs are somewhat necessary, they can be uncommented.